ulid_test.go

```diff
@@ -399,6 +399,48 @@ func TestEntropy(t *testing.T) {
 	}
 }
 
+type flakyEntropyReader struct {
+	data []byte
+	pos  int
+}
+
+func (r *flakyEntropyReader) Read(p []byte) (int, error) {
+	remaining := len(r.data) - r.pos
+	if remaining <= 0 {
+		return 0, io.EOF
+	}
+	n := len(p)
+	if n > 1 {
+		n /= 2
+	}
+	if remaining < n {
+		n = remaining
+	}
+	copy(p, r.data[r.pos:r.pos+n])
+	r.pos += n
+	return n, nil
+}
+
+func TestEntropyRead(t *testing.T) {
+	t.Parallel()
+
+	n, entropyLength := 50, 10
+	r := &flakyEntropyReader{
+		data: make([]byte, n*entropyLength),
+	}
+	rand.Read(r.data)
```

Comment:
How ironic that this could read less than `len(r.data)` :-)

Reply:
Actually it could not, see [docs](https://golang.org/pkg/math/rand/#Read).